### PR TITLE
Sort versions taking into consideration the vcs type

### PR DESCRIPTION
**Pull request by stsewd**
Ref https://github.com/readthedocs/readthedocs.org/issues/6010#issuecomment-519340989

### CODE REVIEW

File: readthedocs/rtd_tests/tests/projects/test_version_sorting.py [hunk 1]
```diff
@@ -65,3 +71,95 @@ def test_sort_alpha(self):
...
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['2.0', '1.10', '1.9', '1.1', '1.0', 'master'],
+            [v.slug for v in sort_version_aware(versions)],
+        )
+
+    def test_sort_svn_trunk(self):
+        identifiers = ['/trunk/', '1.0', '2.0', '1.1', '1.9', '1.10']
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/projects/test_version_sorting.py]: not really sure about this one, but it's as fallback_branch, so :man_shrugging:

and not like we have many of these :eyes:

File: readthedocs/rtd_tests/tests/projects/test_version_sorting.py [hunk 2]
```diff
@@ -65,3 +71,95 @@ def test_sort_alpha(self):
             ['latest', 'carrot', 'banana', 'apple'],
             [v.slug for v in sort_version_aware(versions)],
         )
+
+    def test_sort_git_master(self):
+        identifiers = ['master', '1.0', '2.0', '1.1', '1.9', '1.10']
```
Comment (ericholscher) [inline readthedocs/rtd_tests/tests/projects/test_version_sorting.py]: Could use a test that sorts both `master` and `latest` to confirm their ordering relative to each other as well. Not super important tho.l

Comment (stsewd) [reply readthedocs/rtd_tests/tests/projects/test_version_sorting.py]: Done

File: readthedocs/projects/version_handling.py [hunk 3]
```diff
@@ -49,27 +48,40 @@ def parse_version_failsafe(version_string):
...
     """
     Can be used as ``key`` argument to ``sorted``.
 
     The ``LATEST`` version shall always beat other versions in comparison.
     ``STABLE`` should be listed second. If we cannot figure out the version
     number then we sort it to the bottom of the list.
 
+    If `repo_type` is given, it adds the default "master" version
+    from the VCS (master, default, trunk).
+    This version is highest than LATEST and STABLE.
```
Comment (davidfischer) [inline readthedocs/projects/version_handling.py]:
```suggestion
    This version is sorted higher than LATEST and STABLE.
```

**Pull request merged**